packages/modules/src/BatchedSessionRouterModule.ts

```diff
@@ -0,0 +1,239 @@
+import { Signer, ethers } from 'ethers'
+import MerkleTree from 'merkletreejs'
+import { NODE_CLIENT_URL, Logger } from '@biconomy/common'
+import { hexConcat, arrayify, hexZeroPad, defaultAbiCoder, Bytes } from 'ethers/lib/utils'
+import { keccak256 } from 'ethereumjs-util'
+import {
+  ModuleVersion,
+  CreateSessionDataParams,
+  StorageType,
+  SessionParams,
+  BatchedSessionRouterModuleConfig,
+  ModuleInfo
+} from './utils/Types'
+import {
+  BATCHED_SESSION_ROUTER_MODULE_ADDRESSES_BY_VERSION,
+  SESSION_MANAGER_MODULE_ADDRESSES_BY_VERSION,
+  DEFAULT_SESSION_KEY_MANAGER_MODULE,
+  DEFAULT_BATCHED_SESSION_ROUTER_MODULE
+} from './utils/Constants'
+import { generateRandomHex } from './utils/Uid'
+import { BaseValidationModule } from './BaseValidationModule'
+import { SessionLocalStorage } from './session-storage/SessionLocalStorage'
+import { ISessionStorage, SessionSearchParam, SessionStatus } from './interfaces/ISessionStorage'
+import { SessionKeyManagerModule } from './SessionKeyManagerModule'
+
+export class BatchedSessionRouterModule extends BaseValidationModule {
+  version: ModuleVersion = 'V1_0_0'
+  moduleAddress!: string
+  sessionManagerModuleAddress!: string
+  sessionKeyManagerModule!: SessionKeyManagerModule
+  /**
+   * This constructor is private. Use the static create method to instantiate SessionKeyManagerModule
+   * @param moduleConfig The configuration for the module
+   * @returns An instance of SessionKeyManagerModule
+   */
+  private constructor(moduleConfig: BatchedSessionRouterModuleConfig) {
+    super(moduleConfig)
+  }
+
+  /**
+   * Asynchronously creates and initializes an instance of SessionKeyManagerModule
+   * @param moduleConfig The configuration for the module
+   * @returns A Promise that resolves to an instance of SessionKeyManagerModule
+   */
+  public static async create(
+    moduleConfig: BatchedSessionRouterModuleConfig
+  ): Promise<BatchedSessionRouterModule> {
+    const instance = new BatchedSessionRouterModule(moduleConfig)
+
+    if (moduleConfig.moduleAddress) {
+      instance.moduleAddress = moduleConfig.moduleAddress
+    } else if (moduleConfig.version) {
+      const moduleAddr = BATCHED_SESSION_ROUTER_MODULE_ADDRESSES_BY_VERSION[moduleConfig.version]
+      if (!moduleAddr) {
+        // throw new Error(`Invalid version ${moduleConfig.version}`)
+        Logger.error(`Invalid version ${moduleConfig.version}`)
+        instance.moduleAddress = DEFAULT_BATCHED_SESSION_ROUTER_MODULE
+      } else {
+        instance.moduleAddress = moduleAddr
+      }
+      instance.version = moduleConfig.version as ModuleVersion
```

Comment:
need to review this assignment in case version is invalid and we used default address..

Reply:
ok updated like this 

```
if (moduleConfig.moduleAddress) {
      instance.moduleAddress = moduleConfig.moduleAddress
    } else if (moduleConfig.version) {
      const moduleAddr = BATCHED_SESSION_ROUTER_MODULE_ADDRESSES_BY_VERSION[moduleConfig.version]
      if (!moduleAddr) {
        throw new Error(`Invalid version ${moduleConfig.version}`)
      }
      instance.moduleAddress = moduleAddr
      instance.version = moduleConfig.version as ModuleVersion
    } else {
      instance.moduleAddress = DEFAULT_BATCHED_SESSION_ROUTER_MODULE
      // Note: in this case Version remains the default one
    }
```